libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -365,15 +388,172 @@ class MapboxRouteLineApi(
     }
 
     /**
-     * Used for the vanishing route line feature, this method updates the vanishing point
-     * calculation point based on the route progress. If the vanishing route line feature
-     * was not enabled in [MapboxRouteLineOptions], this method does not need to be called.
+     * Updates the state of the route line based on data in the [RouteProgress] passing a result
+     * to the consumer that should be rendered by the [MapboxRouteLineView].
+     *
+     * If the vanishing route line feature was not enabled in [MapboxRouteLineOptions],
+     * this method does not need to be called.
+     *
+     * Likewise if the style inactive route legs independently feature was not enabled in
+     * [MapboxRouteLineOptions] this method does not need to be called.
      *
      * @param routeProgress a route progress object
+     * @param consumer a consumer for the result of this call
      */
-    fun updateWithRouteProgress(routeProgress: RouteProgress) {
+    fun updateWithRouteProgress(
+        routeProgress: RouteProgress,
+        consumer: MapboxNavigationConsumer<Expected<RouteLineError, RouteLineUpdateValue>>
```

Comment:
It made more sense to me that the method calls the callback when there's something to do, similar to events that are dispatched. Since the caller isn't waiting or otherwise dependent on the callback it is executed at the discretion of the implementation.  I thought it better than returning a "dummy" response that doesn't do anything.